JS/stockSearchMain.js

```diff
@@ -0,0 +1,55 @@
+const basicUrl = "https://stock-exchange-dot-full-stack-course-services.ew.r.appspot.com/api/v3/";
+
+const loader = document.getElementsByClassName("loader-container")[0];
+function turnOffLoader(){
+loader.style.display = "none";
+};
+function turnOnLoader(){
+    loader.style.display = "flex";
+    };
+
+async function askingFreeStockServerRespone(url) {
+    try {
+        const freeStockData = await fetch(url);
+        const jsonFreeStockData = await freeStockData.json();
+    
+    return jsonFreeStockData;
+    }
+    catch(error) {
+       return "Server error"
+    }
+};
+
+async function retrieveCompanyNameAndSymbol(object) {
+    const symbolOfFirstObject = object.symbol;
+    const companyNameOfFirstObject = object.name;
+    const companyNameandSymbol = `${companyNameOfFirstObject} (${symbolOfFirstObject})`;
+    return companyNameandSymbol;
+
+}
+
+async function presentOutput(url) {
+    const resultListElement = document.getElementById("result-list");
+    resultListElement.innerHTML = "";
+    const stockArray = await askingFreeStockServerRespone(url); 
+
+    for (let i = 0; i <= stockArray.length -1; i++) {
+        const companyNameandSymbol = await  retrieveCompanyNameAndSymbol(stockArray[i]);
+        turnOffLoader();             
+        resultListElement.innerHTML += `<li> ${companyNameandSymbol} </li>`;
```

Comment:
Although you code works, DOM manipulation is an action that requires a lot of computing power (DOM access is "expensive" for the browser) and since then it should be always done outside of loop iteration. A better approach would be to aggregate all the new elements inside a parent element that does not live inside the DOM yet, and only after the loop iteration is done you should append the parent element with all the children inside of it. You can also check about createDocumentFragment() (google it!)